### ISSUE TITLE
The juju_ssh_key resource allows creation of malformed SSH keys

### ISSUE THREAD

**Issue by cmisare** [juju/terraform-provider-juju#363]
### Description

The included terraform example creates a malformed SSH key inside of the specified juju model. This malformed SSH key does not seem removable (through either `juju remove-ssh-key` nor by destroying the specified terraform resource).

Juju version: 2.9.46
Terraform version: 1.5.7
Terraform Juju provider version: 0.10.0

### Urgency

Casually reporting

### Terraform Juju Provider version

0.10.0

### Terraform version

v1.5.7

### Terraform Configuration(s)

```terraform
terraform {
  required_providers {
    juju = {
      source  = "juju/juju"
      version = "0.10.0"
    }
  }
}

provider "juju" {}

resource "juju_model" "foo" {
  name = "foo"
}

resource "juju_ssh_key" "bad-ssh-key" {
  model   = juju_model.foo.name
  payload = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAB36uxrTtfdEhfgN5d2P6dt2C3m64t3pBvkn54MFUrw\nkey-label-after-newline"
}
```

### Reproduce / Test

```shell
terraform init
terraform apply
```

```
$ juju list-ssh-keys --model foo
Keys used in model: foo
cf:25:2f:c0:ba:c5:30:63:89:e1:ca:d5:2d:1b:e3:91
Invalid key: key-label-after-newline

$ juju list-ssh-keys --model foo --full
Keys used in model: foo
ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAB36uxrTtfdEhfgN5d2P6dt2C3m64t3pBvkn54MFUrw
Invalid key: key-label-after-newline

$ juju remove-ssh-key --model foo 'cf:25:2f:c0:ba:c5:30:63:89:e1:ca:d5:2d:1b:e3:91'

$ juju remove-ssh-key --model foo 'key-label-after-newline'
cannot remove key id "key-label-after-newline": invalid ssh key: key-label-after-newline
```

```
$ terraform apply -destroy
[...]
juju_ssh_key.bad-ssh-key: Destroying... [id=sshkey:foo:key-label-after-newline]
╷
│ Error: Client Error
│ 
│ Unable to delete ssh key during delete, got error: [invalid ssh key: key-label-after-newline]
╵
```
```


### Debug/Panic Output

_No response_

### Notes & References

_No response_

**Comment by cderici** [comment]
This is related to #267, @cmisare thanks for opening this 👍

**Comment by hmlanigan** [comment]
@Aflynn50 I believe there is a juju bug to fix also
```
$ juju remove-ssh-key --model foo 'key-label-after-newline'
cannot remove key id "key-label-after-newline": invalid ssh key: key-label-after-newline
```

**Comment by Aflynn50** [comment]
This turns out to be a Juju bug ([bug report here](https://bugs.launchpad.net/juju/+bug/2057686)). The fix is in progress and is currently targeted to be released with 3.3.5.
